scripts/prepare_nerfstudio.py

```diff
@@ -0,0 +1,160 @@
+            {
+                "k1": camera_params.k1(),
+                "k2": camera_params.k2(),
+                "k3": 0.0,
+                "k4": 0.0,
+            }
+        )
+
+    out["frames"] = frames
+
+    with open(os.path.join(output_dir, "transforms.json"), "w", encoding="utf-8") as f:
+        json.dump(out, f, indent=4)
+
+    return img_w, img_h
+
+
+def resize_and_save_images_for_nerfstudio(
+    images_dir: str, image_width: int, image_height: int, resized_images_dir: str
+) -> None:
+    """Resizes images to the size used in GTSfM and saves for nerfstudio.
+    Args:
+        images_dir: path to the original images.
+        images_width: int width of resized images
+        images_height: int height of resized images.
+        resized_images_dir: path to directory where the resized images will be saved.
+
+    """
+    image_filenames = os.listdir(images_dir)
+    for image_filename in image_filenames:
+        image_path = os.path.join(images_dir, image_filename)
+        image = io_utils.load_image(image_path)
+        resized_image = images.resize_image(image, image_height, image_width)
+        resized_image_path = os.path.join(resized_images_dir, image_filename)
+        io_utils.save_image(resized_image, resized_image_path)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--results_path", required=True, help="Path to /gtsfm/results/, should contain ba_output/")
+    parser.add_argument(
+        "--images_dir",
+        required=True,
+        help="Path to the directory of images GTSfM used for reconstruction",
+    )
+    parser.add_argument(
+        "--camera_model",
+        default="perspective",
+        help="'perspective' or 'fisheye' corresponding to the OPENCV and OPENCV_FISHEYE camera models at"
+        + " https://github.com/nerfstudio-project/nerfstudio/blob/main/nerfstudio/cameras/cameras.py",
+    )
+    args = parser.parse_args()
+
+    # Create transforms.json, which contains intrinsics and extrinsics
+    results_path = args.results_path
+    cameras_file = os.path.join(results_path, "ba_output", "cameras.txt")
+    images_file = os.path.join(results_path, "ba_output", "images.txt")
+    nerfstudio_input_dir = os.path.join(results_path, "nerfstudio_input")
+    data_dir = os.path.join(results_path, "ba_output")
+    if not os.path.exists(nerfstudio_input_dir):
+        os.makedirs(nerfstudio_input_dir)
+    image_width, image_height = colmap_to_json(
+        data_dir, nerfstudio_input_dir, camera_model=CAMERA_MODELS[args.camera_model]
+    )
+
+    # Resize images (as GTSfM does) to match intrinsics in transforms.json
+    gtsfm_images_dir = args.images_dir
+    nerfstudio_images_dir = os.path.join(nerfstudio_input_dir, "images")
+    if not os.path.exists(nerfstudio_images_dir):
+        os.makedirs(nerfstudio_images_dir)
+    resize_and_save_images_for_nerfstudio(gtsfm_images_dir, image_width, image_height, nerfstudio_images_dir)
```

Comment:
just curious, would it be simpler to just scale the intrinsics, instead of resaving the images on disk, especially if there are 100s of them?

Reply:
Nerfstudio expects the image directory to be in the same directory as their json file, so we have to copy anyway. But you make a good point. It might be better to use the full res images to train nerfs, so I made this default. 
Now we scale intrinsics and copy full res images by default, but there is a flag to use gtsfm's resolution if anyone wants to.

Reply:
Thanks!